src/model/user.rs

```diff
@@ -449,41 +449,43 @@ pub struct User {
     #[serde(rename = "username")]
     pub name: String,
     /// the public flags on a user's account
-    pub public_flags: UserPublicFlags,
+    pub public_flags: Option<UserPublicFlags>,
 }
 
-bitflags! {
-    /// User public flags
-    #[derive(Default, Deserialize, Serialize)]
-    pub struct UserPublicFlags: u32 {
-        /// User's flag as none
-        const NONE = 0b00000000_00000000_00000000_00000000;
+/// User's public flags
+#[derive(Clone, Deserialize, Serialize)]
```

Comment:
This can implement `Copy` as four bytes are very cheap for the processor to copy.

```suggestion
#[derive(Clone, Copy, Deserialize, Serialize)]
```

Reply:
@miqbalrr I can finish this one off is you don't have the time :)

Reply:
Sorry, i wasn't check the notification. i Will finish later